types/wlr_idle_inhibit_v1.c

```diff
@@ -41,8 +44,11 @@ static void idle_inhibitor_handle_surface_destroy(
 		struct wl_listener *listener, void *data) {
 	struct wlr_idle_inhibitor_v1 *inhibitor =
 		wl_container_of(listener, inhibitor, surface_destroy);
+	struct wl_resource *resource = inhibitor->resource;
 
-	wl_resource_destroy(inhibitor->resource);
+	/* make resource inert */
+	idle_inhibitor_v1_handle_resource_destroy(resource);
```

Comment:
Hmmm. Please use the template from CONTRIBUTING.md, I don't like how generic `wl_resource`s are passed around.

Reply:
Can do for this one, but the `idle_inhibit_manager_v1_handle_resource_destroy` does not allocate its own type (only set its link + user data) so I don't really have a choice there.